pkg/webhookadmission/utils.go

```diff
@@ -0,0 +1,141 @@
+package webhookadmission
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"time"
+
+	"github.com/portworx/sched-ops/k8s"
+	log "github.com/sirupsen/logrus"
+	hook "k8s.io/api/admissionregistration/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	webhookName        = "webhook.stork.libopenstorage.io"
+	storkService       = "stork-service"
+	storkNamespace     = "STORK-NAMESPACE"
+	defaultNamespace   = "kube-system"
+	storkServiceAccess = "stork-service.kube-system.svc"
+)
+
+// CreateMutateWebhook create new webhookconfig for stork if not exist already
+func CreateMutateWebhook(caBundle []byte, ns string) error {
+	path := "/mutate"
+	webhook := hook.Webhook{
+		Name: webhookName,
+		ClientConfig: hook.WebhookClientConfig{
+			Service: &hook.ServiceReference{
+				Name:      storkService,
+				Namespace: ns,
+				Path:      &path,
+			},
+			CABundle: caBundle,
+		},
+		Rules: []hook.RuleWithOperations{
+			{
+				Operations: []hook.OperationType{hook.Create},
+				Rule: hook.Rule{
+					APIGroups:   []string{"apps", ""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"deployments", "statefulset", "pods"},
+				},
+			},
+		},
+	}
+	req := &hook.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: storkAdmissionController,
+		},
+		Webhooks: []hook.Webhook{webhook},
+	}
+	_, err := k8s.Instance().CreateMutatingWebhookConfiguration(req)
+	if !k8serr.IsAlreadyExists(err) && err != nil {
+		log.Errorf("unable to create mutate webhook: %v", err)
+		return err
+	}
+	log.Debugf("Created mutating webhook configuration: %v", webhookName)
+	return nil
+}
+
+// GenerateCertificate Self Signed certificate , returns x509 cert
+// and priv key in PEM format
+func GenerateCertificate() ([]byte, []byte, error) {
+	var err error
+	pemCert := &bytes.Buffer{}
+
+	// generate private key
+	priv, err := ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
+	if err != nil {
+		log.Errorf("error generating crypt keys: %v", err)
+		return nil, nil, err
+	}
+
+	// create certificate
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName: storkServiceAccess,
```

Comment:
Can't use `storkServiceAccess` since it has the namespace hardcoded to `kube-system`, need to use the namespace where stork is started.